MapzenSDK/MZMapViewController.swift

```diff
@@ -1177,21 +1071,29 @@ open class MZMapViewController: UIViewController, LocationManagerDelegate {
 extension MZMapViewController : TGMapViewDelegate, TGRecognizerDelegate {
   
   //MARK : TGMapViewDelegate
-  
-  open func mapView(_ mapView: TGMapViewController, didLoadSceneAsync scene: String) {
-    // if we loaded a house style scene looks something like: file:///var/containers/Bundle/Application/FAFA232A-1190-40CB-9391-7C9F44B51076/ios-sdk.app/housestyles.bundle/bubble-wrap/bubble-wrap-style-more-labels.yaml
-    guard let pathComponents = URL.init(string: scene)?.pathComponents else {
-      onStyleLoadedClosure = nil
+
+  open func mapView(_ mapView: TGMapViewController, didLoadScene sceneID: Int32, withError sceneError: Error?) {
+
+    //We only want to call back on the latest scene load - so we gate here to make sure we only call back on the latest.
+    //TODO: For 2.0 we should pass the Error along in the callback block.
+    if sceneID != latestSceneId {
       return
     }
-    // if we have path components, grab the last two (ie. bubble-wrap & bubble-wrap-style-more-labels.yaml), strip ".yaml" and check for existence in styles map
-    let sceneStyle = (pathComponents[pathComponents.count-2] + "/" + pathComponents.last!).replacingOccurrences(of: ".yaml", with: "")
-    guard let style = styles[sceneStyle] else {
-      onStyleLoadedClosure = nil
-      return
+    restoreBuiltInMarkers()
```

Comment:
We might want to move this to the top of the method and always restore built in markers so that other map methods like enabling transit overlays etc. don't cause built in markers to disappear.

Reply:
My only concern there is we don't want to be restoring markers if we have another scene load coming through, which is why this happens underneath the `sceneId` check. We'd basically be wasting cycles. I can test the transit overlay scenario more but I don't recall running into that problem.

Reply:
Sorry for jumping in late, but should sdk provide interface for the user to decide if they want to retain the markers from a previous scene load when a new scene load is requested?